### PR TITLE
Fix Numba CI issues

### PR DESCRIPTION
**Pull request by awolant**
Signed-off-by: Albert Wolant <awolant@nvidia.com>

## Category:

Bug fix

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: qa/TL0_python-self-test_xavier/test_body.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ test_py_with_framework() {
         "'mixed'"
         "video\("
         "caffe"
+        "numba"
```
[the remark this answers is not in the document]
Comment (mzient) [reply qa/TL0_python-self-test_xavier/test_body.sh:14]: :+1:

File: dali/python/nvidia/dali/plugin/numba/experimental.py [hunk 2]
```diff
@@ -311,6 +316,11 @@ def __call__(self, *inputs, **kwargs):
         return outputs[0] if len(outputs) == 1 else outputs
 
     def __init__(self, run_fn, out_types, in_types, outs_ndim, ins_ndim, setup_fn=None, device='cpu', batch_processing=False, blocks=None, threads_per_block=None, **kwargs):
+        toolkit_version = cuda.runtime.get_version()
+        driver_version = cuda.driver.driver.get_version()
+
+        assert toolkit_version <= driver_version, f"Environment is not compatible with Numba GPU operator. Driver version is {driver_version} and CUDA Toolkit version is {toolkit_version}. Driver can not be older than the CUDA Toolkit"
```
Comment (klecki) [inline dali/python/nvidia/dali/plugin/numba/experimental.py:322]: As always, one can disable asserts, it should be a Runtime Error raised here.

File: dali/test/python/test_operator_numba_func_gpu.py [hunk 3]
```diff
@@ -83,11 +89,13 @@ def test_numba_func():
...
 
 
+@with_setup(check_env_compatibility)
 @raises(AssertionError, glob='Currently batch processing for GPU is not supported.')
 def test_batch_processing_assertion():
     _testimpl_numba_func([(10, 5)], np.uint8, set_all_values_to_255_sample, [dali_types.UINT8], [dali_types.UINT8], [2], [2], [1, 1, 1], [10, 5, 1], None, True, [np.full((10, 5), 255, dtype=np.uint8)])
 
 
+@with_setup(check_env_compatibility)
 @raises(RuntimeError, regex="Shape of input [0-9]+, sample at index [0-9]+ doesn't match the shape of first sample")
```
Comment (klecki) [inline dali/test/python/test_operator_numba_func_gpu.py:99]: Shouldn't all numba test be tagged with `@attr(numba)`? Or we run this particular file only in specific job? I assume it is picked up by the test_operator_* job?

Comment (awolant) [reply dali/test/python/test_operator_numba_func_gpu.py:99]: They were not tagged in CPU variant, so they are not tagged for GPU as well. 

Actually, the only place where this tag is used is in the functional API test.

File: dali/python/nvidia/dali/plugin/numba/experimental.py (hunk 2 above)
Comment (awolant) [reply dali/python/nvidia/dali/plugin/numba/experimental.py:322]: Done

File: dali/test/python/test_operator_numba_func_gpu.py (hunk 3 above)
Comment (klecki) [reply dali/test/python/test_operator_numba_func_gpu.py:99]: I am a bit surprised, as we have tests that disallow numba, but maybe we just assume it is fine to test it in the operators_* job.

**Comment by awolant** [comment]
!build

**Pull request merged**